### ISSUE TITLE
How to use the same split like TFix evaluation?

### ISSUE THREAD

**Issue by smith-co** [eth-sri/TFix#19]
Congratulations for your fantastic work and great repo.

Can you please help me to understand how to get the exact same split for train and test dataset that you have used for TFix?

In the paper you mentioned:

**CLEAN TEST:**
- Our final dataset consists of `104, 804` samples. Detailed statistics are shown in Table 1, and the description for each error type can be found in (esl, 2021). 
- To a create train-test split, we randomly selected 10% of the samples for each error type as the test set (we call it `clean test`). The remaining was further split into `90%` for fine-tuning and `10% for` validation.

**RANDOM TEST:**
- Besides clean test, we test how TFix generalizes to errors where the original fix was complex or noisy and therefore dropped in our data collection procedure. 
- To this end, we assembled another test set called random test consisting of all fixable errors in the GitHub commits except for the ones used for fine-tuning and validation. 

However it is not clear to me how can I get exact same test and train split that you have used in your evaluation for TFix? Can you point me how from the following files I can generate the exact same split like your evaluation. 

data_autofix_tracking_repo_specific_filtered.json
data_autofix_tracking_eslint_filtered.json
data_autofix_tracking_eslint_final.json
data_autofix_tracking_repo_specific_final.json

Essentially I would really appreciate if you can point me how from these jsons I can use the exact same split?

**Comment by BBerabi** [comment]
Hi @smith-co,

Thank you for your kind words, and nice to see your interest in our paper.

The datasets are split on the fly in the code. Considering this design now, I can agree that it would be a better design choice to provide separate files for each test split. However, you can still accomplish this.

### CLEAN TEST

We fix random seeds so that one gets the same splits for training validation and testing in each run. The split is done on the fly during data preprocessing. The whole data is given to the function `create_data` in the main training script. Please look at this line: https://github.com/eth-sri/TFix/blob/main/hf_transformers/tfix_training.py#L79.

Essentially, this function returns the inputs, labels and some metadata information for each split. You can again see it in the link. So, my suggestion for you would be  to copy the code until that point, and after you have inputs, labels and metadata information for each split, you can save it into a file. You will probably need to play around with the code and data to understand the workflow and what each variable contains. It also depends on in which schema you want to save the files. After you have the files, you can integrate them into your workflow.

If you want more details on how the split in the code is done, you can track down the code in `create_data`. For example, it would lead you to this other function. https://github.com/eth-sri/TFix/blob/main/hf_transformers/prepare_data.py#L29. As you can see, the seed is fixed in the function argument with a default value.

Note that the clean test corresponds to files with the suffix `_final.json`. Again you can look at what the code reads. (https://github.com/eth-sri/TFix/blob/main/hf_transformers/tfix_training.py#L62)

### RANDOM TEST
 
For these tests, you do not need to apply to split. Because as you quoted from our paper, these samples were cleaned during the data collection procedure because the extracted fixed code was considered noisy or wrong. So, all of these samples were used as test samples in `RandomTest`. No samples from these datasets made it into training.

Note that the random test corresponds to files with the suffix `_filtered.json`. Filtered means, in this context, cleaned.

I hope this clarifies everything!

Best,
Berkay

**Comment by ghost** [comment]
I am closing this issue as it is well answered and has been inactive for awhile. If you have other questions, please open a new issue. Thank you!
